pkg/util/fileutil/writer_test.go

```diff
@@ -0,0 +1,340 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fileutil
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"sigs.k8s.io/secrets-store-csi-driver/pkg/util/tmpdir"
+	"sigs.k8s.io/secrets-store-csi-driver/provider/v1alpha1"
+)
+
+func TestValidatePath_Success(t *testing.T) {
+	cases := []struct {
+		name string
+		path string
+	}{
+		{
+			name: "valid 1",
+			path: "i/am/well/behaved.txt",
+		},
+		{
+			name: "valid 2",
+			path: "keepyourheaddownandfollowtherules.txt",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			if err := validatePath(tc.path); err != nil {
+				t.Errorf("unexpected failure: %v", err)
+			}
+		})
+	}
+}
+
+func TestValidatePath_Error(t *testing.T) {
+	maxPath := strings.Repeat("a", maxPathLength+1)
+	maxFile := strings.Repeat("a", maxFileNameLength+1)
+
+	cases := []struct {
+		name string
+		path string
+	}{
+		{
+			name: "max path length",
+			path: maxPath,
+		},
+		{
+			name: "max file length",
+			path: maxFile,
+		},
+		{
+			name: "absolute failure",
+			path: "/dev/null",
+		},
+		{
+			name: "reserved path",
+			path: "..sneaky.txt",
+		},
+		{
+			name: "contains doubledot 1",
+			path: "hello/there/../../../../../../etc/passwd",
+		},
+		{
+			name: "contains doubledot 2",
+			path: "hello/../etc/somethingbad",
+		},
+		{
+			name: "empty",
+			path: "",
+		},
```

Comment:
Can we also add a test to cover https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/481/files#diff-08554986af7961dac0cae0b9d5a8abac412b1a064d2f46853dcd822f6dd1f6f6R114-R116

Reply:
This looks good! Thanks for adding.